services/web/server/src/simcore_service_webserver/computation_subscribe.py

```diff
@@ -84,55 +79,96 @@ async def rabbitmq_consumer(app: web.Application) -> AsyncIterator[None]:
-    connection = await aio_pika.connect_robust(
-        f"{rabbit_broker}" + f"?name={__name__}_{id(app)}",
-        client_properties={"connection_name": "webserver read connection"},
-    )
-
-    channel = await connection.channel()
-    await channel.set_qos(prefetch_count=1)
-
-    async def _connect_to_queue(
-        name: str,
-        handler: Callable[[aio_pika.IncomingMessage, Application], Awaitable[None]],
-        no_ack: bool,
-    ):
-        exchange = await channel.declare_exchange(name, aio_pika.ExchangeType.FANOUT)
-        # Declaring queue
-        queue = await channel.declare_queue(
-            f"webserver_{id(app)}_{name}",
-            exclusive=True,
-            arguments={"x-message-ttl": 60000},
+    async def get_connection() -> aio_pika.Connection:
+        return await aio_pika.connect_robust(
+            f"{rabbit_broker}" + f"?name={__name__}_{id(app)}",
+            client_properties={"connection_name": "webserver read connection"},
         )
-        # Binding the queue to the exchange
-        await queue.bind(exchange)
-        # pass the handler
-        await queue.consume(
-            functools.partial(handler, app=app),
-            exclusive=True,
-            no_ack=no_ack,
+
+    connection_pool = aio_pika.pool.Pool(get_connection, max_size=2)
+
+    async def get_channel() -> aio_pika.Channel:
+        async with connection_pool.acquire() as connection:
+            channel = await connection.channel()
+            # Finding a suitable prefetch value is a matter of trial and error
+            # and will vary from workload to workload. Values in the 100
+            # through 300 range usually offer optimal throughput and do not
+            # run significant risk of overwhelming consumers. Higher values
+            # often run into the law of diminishing returns.
+            # Prefetch value of 1 is the most conservative. It will significantly
+            # reduce throughput, in particular in environments where consumer
+            # connection latency is high. For many applications, a higher value
+            # would be appropriate and optimal.
+            await channel.set_qos(prefetch_count=100)
+            return channel
+
+    channel_pool = aio_pika.pool.Pool(get_channel, max_size=10)
+
+    async def exchange_consumer(
+        exchange_name: str,
+        parse_handler: Callable[[web.Application, Dict[str, Any]], Awaitable[None]],
+        consumer_kwargs: Dict[str, Any],
+    ):
+        async with channel_pool.acquire() as channel:
+            exchange = await channel.declare_exchange(
+                exchange_name, aio_pika.ExchangeType.FANOUT
+            )
+            # Declaring queue
+            queue = await channel.declare_queue(
+                f"webserver_{id(app)}_{exchange_name}",
```

Comment:
I think we should avoid ``id(app)`` in the  naming and use as well  with the new hostname that we pass to the container . it will be more traceable than this random id. It applies to all

Reply:
since there might be (in some near future), more than one webserver on the same host I will keep the id as well.
but I did add some more meaningfull name as well.